api/tests/unit/domain/models/CertificationCourse_test.js

```diff
@@ -34,21 +34,53 @@ describe('Unit | Domain | Models | CertificationCourse', () => {
     });
   });
 
-  describe('#correctBirthdate', () => {
+  describe('#uncancel #isCancelled', function() {
+
+    it('should uncancel a certification course', function() {
+      // given
+      const certificationCourse = domainBuilder.buildCertificationCourse({
+        isCancelled: true,
+      });
+
+      // when
+      certificationCourse.uncancel();
+
+      // then
+      expect(certificationCourse.toDTO().isCancelled).to.be.false;
+    });
+
+    describe('when certification course is already uncancelled', function() {
+      it('should not change isCancelled value', function() {
+        // given
+        const certificationCourse = domainBuilder.buildCertificationCourse({
+          isCancelled: true,
+        });
+
+        // when
+        certificationCourse.uncancel();
+        certificationCourse.uncancel();
+
+        // then
+        expect(certificationCourse.toDTO().isCancelled).to.be.false;
+      });
```

Comment:
En fait ici on cherche à montrer l'idem potence. Une manière plus élégante pour le faire, à mon sens, serait ceci :
```suggestion
      it('should be an idempotent operation', function() {
        // given
        const certificationCourse = domainBuilder.buildCertificationCourse({
          isCancelled: true,
        });

        // when
        certificationCourse.uncancel();
        const firstTimeDTO = certificationCourse.toDTO();
        certificationCourse.uncancel();
        const secondTimeDTO = certificationCourse.toDTO();

        // then
        expect(firstTimeDTO).to.deep.equal(secondTimeDTO);
      });
```

Reply:
D'acc